synfig-studio/src/gui/workarea.cpp

```diff
@@ -1492,10 +1482,29 @@ WorkArea::on_drawing_area_event(GdkEvent *event)
 	        break;
 		}
 		case DRAG_GUIDE: {
-			if(curr_guide_is_x)
-				*curr_guide = mouse_pos[0];
-			else
-				*curr_guide = mouse_pos[1];
+			if (curr_guide->isVertical) {//should then be replaced by checking current guides isVertical
+				if((!rotate_guide && (synfig::Angle::deg(curr_guide->angle).get() == 90))){// case 1: unrotated ruler (vertical)
+					curr_guide->point[0] = mouse_pos[0];
+				} else if (!rotate_guide) { //case 2: rotated ruler being moved
+					float center_x_new= (mouse_pos[0])+((-mouse_pos[1]+curr_guide->point[1])/(tan(curr_guide->angle.get())));
+					curr_guide->point[0] = center_x_new;
+				}
```

Comment:
Shouldn't both branches give the same result if `curr_guide->angle` is 90°?

Reply:
well, currently yes and no. they both behave similarly in a way that if both are 90 degrees they can both be dragged left and right. but for horizontal rulers this makes them lose their center of rotations, they are no longer able to be rotated unless dragged back into correct place. this goes back to the fact that the centers of rotations axis for horizontal rulers and vertical rulers are different (line passing through center of width and center of height).

Reply:
When a new guide (not the ruler, right?) is created, it will be horizontal or vertical, right? With its initial c.o.r set as the image center, isn't it? I don't even know how to change the c.o.r by mouse (not necessary btw).
So I don't really get the point about this variable.

Reply:
I don't know maybe i'm not getting what you mean exactly. But no not always, actually the only case they both have the same c.o.r is at the exact center of the canvas when for example creating by a click not dragging from the ruler. Even then when they are being moved the c.o.r moves differently one moves on x axis other on y (this was to give more flexibility instead of say for example both move only on x).

Reply:
Ok, I think I kind of get what your reasoning is (correct me if i'm wrong). You are saying once the rulers are rotated theoretically only the new mouse point is needed (slope is stored) to draw out the guides. This is true for the the guide line equation yes but for the c.o.r it is not we need to know if this ruler is vertical or not to be able to determine how we get the c.o.r. For example if it is we look for intersection with horizontal line passing through half the screen height, for horizontal it is the vertical line passing through half screen width.

So if isVertical isn't stored how would it be determined if this guide is vertical or horizontal to determine the c.o.r ?... Actually I think I kind of get why you think that. Maybe you are thinking horizontal guides become vertical when they are 90 degress so then the isVertical would be outdated info stored ? but this isn't the case a horizontal guide (a guide created from the horizontal ruler) is always a horizontal guide (same c.o.r axis a.lways) even if it becomes vertical (90 degress).

so in a nutshell basically isVertical is used for determining the c.o.r axis vertical or horizontal. A guide created from a vertical ruler is a vertical guide through out its usage and never becomes a horizontal guide even if it look horizontal. 

I think I typed a lot xD, hopefully though now I understand you ?... Also I suggest you can do a little test that will make it realy clear what I mean  if it is possible.
1- create two guides
2- edit their angle to 45 
3- move both to any direction (right for example)
4- notice how the c.o.r (blue plus sign) is moving 

p.s. I will see if I can record the example instead I think that will make it easier.

Reply:
An alternative maybe to this method that I just thought of now could be to discard this vertical and horizontal concept altogether and just determine the c.o.r depending on the ruler location. Having it always be the center of the distance between the two furthest guide points (the ones intersecting a canvas axis). 
pros: c.o.r is found in a unified way... always center of the guide length.
cons:
- I think this would  be heavier in calculations as you would be adding the burden of calculating a new c.o.r every time the guide is moved (rotating would keep the ).
-  I just thought of this and I think this just makes the whole idea useless. in certain places the guide c.o.r would be different depending on its angle which would be a huge mess.

Reply:
@rodolforg  this is what I meant. For some reason mouse wasn't recorded but basically I was just moving them right then left.

[demoingcor.webm](https://user-images.githubusercontent.com/100296264/221022139-d9366993-0949-4f83-942b-9441469bbb45.webm)

Reply:
isVertical removed :)